source/api_cc/src/DeepPot.cc

```diff
@@ -750,14 +785,21 @@ compute (std::vector<ENERGYTYPE> &		all_energy,
         nlist_data.shuffle(atommap);
 	nlist_data.make_inlist(nlist);
     }
-    int ret = session_input_tensors (input_tensors, dcoord_, ntypes, datype_, dbox, nlist, fparam, aparam, atommap, nghost, ago);
 
     all_energy.resize (numb_models);
     all_force.resize (numb_models);
     all_virial.resize (numb_models);
-    assert (nloc == ret);
+
     for (unsigned ii = 0; ii < numb_models; ++ii) {
-        run_model (all_energy[ii], all_force[ii], all_virial[ii], sessions[ii], input_tensors, atommap, nghost);
+      if (dtype == tensorflow::DT_DOUBLE) {
+        int ret = session_input_tensors <double> (input_tensors, dcoord_, ntypes, datype_, dbox, nlist, fparam, aparam, atommap, nghost, ago);
```

Comment:
The `session_input_tensors` should be placed out of the for loop, because all models share the same inputs, and we only need to make the input tensors once.